lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
But how user will know is connection hanged with no reply (they may want to send HTTP 408 Request Timeout) or it is just closed after all activities are finished successfully? That is not obvious for API user. Somehow successfully ended not keep-alive connection does not generate timeout events, but successfully ended keep-alive generates timeout events. Now we have connection timeot and keep-alive timeout, so connection timeout indicates ```hang``` state, but keep-alive timeout is a different thisn I suppose, @indutny